jujugui/static/gui/src/app/init/charmstore-user.js

```diff
@@ -0,0 +1,58 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const CharmstoreUserMixin = (superclass) => class extends superclass {
+
+  /**
+    Takes a macaroon and stores the user info (if any) in the app.
+    @param {String} service The service the macaroon comes from.
+    @param {String} macaroon The base64 encoded macaroon.
+    @param {Boolean} rerenderProfile Rerender the user profile.
+    @param {Boolean} rerenderBreadcrumb Rerender the breadcrumb.
+   */
+  storeUser(service, rerenderProfile, rerenderBreadcrumb) {
+    const callback = (error, auth) => {
+      if (error) {
+        console.error('Unable to query user information', error);
+        return;
+      }
+      if (auth) {
+        this.users[service] = auth;
```

Comment:
As mentioned, since this relies on this.users, then this.users (or better, this._csUsers) should be defined here.
Otherwise mixins can easily become a nightmare in which you jump here and there trying to find where the state comes from.